src/context.cc

```diff
@@ -458,13 +458,15 @@ FilterStatus ContextBase::onUpstreamData(uint32_t data_length, bool end_of_strea
 
 void ContextBase::onDownstreamConnectionClose(CloseType close_type) {
   if (!isFailed() && wasm_->on_downstream_connection_close_) {
+    wasm_->addAfterVmCallAction([this]() { this->stop_iteration_ = false; });
```

Comment:
Wouldn't it be easier to just add this to `DeferAfterCallActions`'s destructor, so that it's called for every function?